packages/@aws-cdk/cdk/lib/core/construct.ts

```diff
@@ -447,6 +458,24 @@ export class ConstructNode {
     return Array.from(ret);
   }
 
+  /**
+   * Triggers each aspect to invoke visit
+   */
+  private invokeAspects(): void {
+    const descendants = this.findAll();
+    const aspectsToVisit = this.visitableAspects();
+    aspectsToVisit.forEach( aspect => {
+      descendants.forEach( member => aspect.visit(member));
+      this.invokedAspects.push(aspect);
+    });
+  }
+
+  private visitableAspects(): IAspect[] {
+    return this.aspects.filter( aspect => {
+      return aspect.visitType === AspectVisitType.Multiple ||
+        !this.invokedAspects.includes(aspect);
+    });
+  }
```

Comment:
The functions above here could be extracted into a class for composition. They were not very complicated so I left them here. If we went to a class we could likely also grab the two arrays I've added to construct and move those into the class. @rix0rrr @eladb -- let me know if you have a strong opinion here.